manticore/platforms/linux.py

```diff
@@ -232,6 +236,7 @@ def connect(self, peer):
             peer.peer = self
 
     def read(self, size):
+        assert not self.is_empty(), "Attempting read() from empty Socket buf"
```

Comment:
i don't think an assert is the right way to handle this error case? i think we should handle it by returning an empty list, not AssertionError